frontend/packages/operator-lifecycle-manager/src/components/operand.tsx

```diff
@@ -167,8 +167,77 @@ export const OperandTableHeader = () => {
   ];
 };
 
+export enum OperatorStatusType {
+  phase = 'phase',
+  status = 'status',
+  state = 'state',
+  conditions = 'conditions',
+}
+
+export const OperatorStatusTypeText = {
+  [OperatorStatusType.phase]: 'Phase:',
+  [OperatorStatusType.status]: 'Status:',
+  [OperatorStatusType.state]: 'State:',
+  [OperatorStatusType.conditions]: 'Condition:',
+};
+
+export type ConditionType = {
+  type: string;
+  status: string;
+  reason?: string;
+  message?: string;
+  lastUpdateTime?: string;
+  lastTransitionTime?: string;
+};
+
+const DeriveIconFromStatusCondition = (conditionStatus: [ConditionType]) => {
+  let statusIcon;
+  if (!_.isEmpty(conditionStatus)) {
+    if (_.has(conditionStatus[0], 'type')) {
+      statusIcon = <Status status={conditionStatus[0].type} />;
+    }
+  }
+  return statusIcon;
+};
+
+const DeriveIconFromStatus = (status: string) => {
+  let statusIcon;
+  if (!_.isEmpty(status)) {
+    statusIcon =
+      status === 'Running' ? <SuccessStatus title={status} /> : <Status status={status} />;
+  }
+  return statusIcon;
+};
+
+export const OperandStatusIconAndText: React.FunctionComponent<OperandStatusIconAndTextProps> = ({
```

Comment:
Nice, I like the cleanup.  I may take it one step further (since `found` returns a value & you can use `const`):

```typescript
export const OperandStatusIconAndText: React.FunctionComponent<OperandStatusIconAndTextProps> = ({
    statusObject,
}) => {
    const empty = <div className="text-muted">Unknown</div>;
    return _.isEmpty(statusObject)
       ? empty
       : _.find(Object.keys(OperatorStatusType), (key) => {
            if (_.has(statusObject, key)) {
                const status = statusObject[key];
                const statusIcon =
                    key === OperatorStatusType.conditions
                    ? DeriveIconFromStatusCondition(status)
                    : DeriveIconFromStatus(status);
                return statusIcon && <span className="co-icon-and-text">
                    {OperatorStatusTypeText[key]}&nbsp;{statusIcon}
                </span>            
            }
        }); 
};
```

Reply:
hi @benjaminapetersen, this isn't compiling for me...are we just trying to reduce code size here? I'm getting an error: ```Type '({ statusObject, }: PropsWithChildren<OperandStatusIconAndTextProps>) => string | Element' is not assignable to type 'FunctionComponent<OperandStatusIconAndTextProps>'.```

Reply:
@benjaminapetersen also, the logic is incorrect with the above logic. What is it you don't like about the original code, maybe I work with you to get it where you want to be?

Reply:
Ha yup. Realized after I left for the weekend, `_.find` only does a find, return has to be boolean, and it will return the data point, that you would have to manipulate after.

Reply:
Just thinking about 
```
return (iconAndText = 
```
is really 2 things, returning `truthy` to short circuit the find, but also the side effect of setting the variable.

Reply:
Might be closer to what I'm thinking (but didn't run to test):

```typescript 
export const OperandStatusIconAndText: React.FunctionComponent<OperandStatusIconAndTextProps> = ({
    statusObject,
}) => {
    const empty = <div className="text-muted">Unknown</div>;
    
    const filled = (status, icon) => (<span className="co-icon-and-text">
        {status}&nbsp;{icon}
    </span>)    

    const foundStatus = _.find(Object.keys(OperatorStatusType), (key) => {
        return _.has(statusObject, key);
    });

    const icon = foundStatus == OperatorStatusType.conditions 
        ? DeriveIconFromStatusCondition(status)
        : DeriveIconFromStatus(status);

    return _.isEmpty(statusObject)
        ? empty 
        : filled(OperatorStatusTypeText[foundStatus], icon);
};
```